### PR TITLE
[CORE] Move driver/executor endpoint to CH backend

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

The driver/executor endpoint only used for ch backend but live in gluten-core, this pr move these to ch backend. After this change, we can also remove the `BroadcastApi`,  `BroadcastApi` is used to handle some special code for ch backend broadcast join with driver endpoint and we can put these code to ch broadcast join.

## How was this patch tested?

Pass CI

### CODE REVIEW

File: gluten-core/pom.xml [hunk 1]
```diff
@@ -24,10 +24,6 @@
       <artifactId>gluten-ui</artifactId>
       <version>${project.version}</version>
     </dependency>
-    <dependency>
-      <groupId>com.github.ben-manes.caffeine</groupId>
-      <artifactId>caffeine</artifactId>
-    </dependency>
```
Comment (ulysses-you) [inline gluten-core/pom.xml:30]: Required for CH backend, so Velox backend can get rid of it.

**Comment by ulysses-you** [comment]
cc @zhztheplayer @PHILO-HE @binmahone @zzcclp thank you

**Comment by zhztheplayer** [comment]
So this lifecycle feature is not really that common for all backends? (cc @baibaichen)

**Comment by baibaichen** [comment]
> So this lifecycle feature is not really that common for all backends? (cc @baibaichen)

@ulysses-you @zhztheplayer 

We use such mechanism to free memory allocated at Broadcast case. Don't Velox need it?

**Comment by ulysses-you** [comment]
@zhztheplayer @baibaichen IIUC, Velox backend follows the vanilla Spark broadcast exchange way. I'm not sure why do we need additional lifecycle.

**Comment by zhztheplayer** [comment]
> @zhztheplayer @baibaichen IIUC, Velox backend follows the vanilla Spark broadcast exchange way. I'm not sure why do we need additional lifecycle.

I assume CH backend uses the feature for storing broadcast data at process wise. If so it's OK to me to move the code if we don't tend do the same in Velox backend.

**Review by zhztheplayer — DISMISSED** [review]
+0.9 since we can still pull up some code once Velox backend requires that feature

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/ListenerApi.scala [hunk 2]
```diff
@@ -16,11 +16,11 @@
...
-import org.apache.spark.SparkConf
+import org.apache.spark.{SparkConf, SparkContext}
 
 trait ListenerApi {
-  def onDriverStart(conf: SparkConf): Unit = {}
+  def onDriverStart(sc: SparkContext, conf: SparkConf): Unit = {}
   def onDriverShutdown(): Unit = {}
-  def onExecutorStart(conf: SparkConf): Unit = {}
+  def onExecutorStart(executorID: String, conf: SparkConf): Unit = {}
   def onExecutorShutdown(): Unit = {}
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/backendsapi/ListenerApi.scala:25]: nit

```suggestion
  def onDriverStart(sc: SparkContext, pc: PluginContext): Unit = {}
  def onDriverShutdown(): Unit = {}
  def onExecutorStart(pc: PluginContext): Unit = {}
  def onExecutorShutdown(): Unit = {}
```

to keep up with Spark's plugin APIs

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala [hunk 3]
```diff
@@ -70,4 +70,6 @@ trait TransformerApi {
   def getNativePlanString(substraitPlan: Array[Byte], details: Boolean): String
 
   def packPBMessage(message: Message): Any
+
+  def invalidateSQLExecutionResource(executionId: String): Unit = {}
```
Comment (zhztheplayer) [inline gluten-core/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala:74]: nit

Probably add a comment saying that the method is only for testing so far?

Comment (ulysses-you) [reply gluten-core/src/main/scala/org/apache/gluten/backendsapi/TransformerApi.scala:75]: addressed

**Review by zhztheplayer — APPROVED** [review]
Thanks!

File: gluten-core/src/main/scala/org/apache/gluten/backendsapi/ListenerApi.scala (hunk 2 above)
Comment (ulysses-you) [reply gluten-core/src/main/scala/org/apache/gluten/backendsapi/ListenerApi.scala]: addressed

**Review by zhztheplayer — APPROVED** [review]

**Pull request merged**